app/javascript/packs/application.js

```diff
@@ -18,7 +18,8 @@ import 'bootstrap';
 // Fontawesome: https://fontawesome.com/
 import "@fortawesome/fontawesome-free/js/all";
 
-import * as L from "leaflet"
+import * as L from "leaflet";
```

Comment:
## The Problem

It seems like leaflet is imported globally. This means that we load and parse the JS that is required for leaflet on every page, even if we don't need it.

How to verify that:
1. start the application (obviously)
2. open the application in an incognito tab (not on the map but e.g. the landing page)
3. open the developer console in the browser
4. enter `L` -> you can see that leaflet has been loaded and parsed

## Is this really a problem?
I don't really know. Probably not. But we should at least be aware of it.

## What can we do about it?
I would expect from a framework like ruby on rails that there is a recommended way of importing libraries that are only required on one or a handful of pages. - I couldn't find any with a quick google search.

Some keywords would be `code splitting` and `lazy loading`.

TLDR:
If there is a good ruby way to do this, then we should. Else don't waste your time on this.

Reply:
We were also wondering about that because vs code says `'L' is declared but its value is never read.`.

Reply:
Das Semikolon ist bei mir aber schon da....

Reply:
> Is this really a problem?

Good that you noticed and thought about this! If a user ever navigates to a page using `Leaflet` the code would need to be loaded anyways. The browser should cache the response, so the JS won't be loaded from the server with every request. The only time that making the code "page specific" would make a difference, is when a user only visits pages without a map, right? 

I also like the following StackOverflow discussion related to this question: https://stackoverflow.com/questions/59493803/

Reply:
leaflet is a small library that doesn't run any computation-intensive functions on initialisation. Therefore this isn't a problem.